packages/firestore/src/local/persistence_promise.ts

```diff
@@ -250,18 +225,15 @@ export class PersistencePromise<T> {
    * PersistencePromises to resolve.
    */
   static forEach<K>(
-    collection: Iterable<K>,
-    f: (item: K) => PersistencePromise<void>
+    // tslint:disable-next-line:no-any Accept any kind of `forEach`.
+    collection: { forEach: ((cb: ((k: K, ...args: any[]) => void)) => void) },
+    // tslint:disable-next-line:no-any Accept any kind of callback.
+    f: (k: K, ...args: any[]) => PersistencePromise<void>
```

Comment:
Uh...  these type gymnastics are pretty crazy.  Seems like there has to be a better way?  

Maybe just define our own Iterable type (with a normal method instead of Symbol.iterator)?

FWIW I wouldn't be surprised if doing `...args` in an inner loop is a mildly bad practice for performance reasons.  Though I'd have to look at the generated code to be sure.

Reply:
I changed the type a bit to remove the varargs. This gets us saner types and a bit more type safety (see the related changes in the last commit).

The generated code uses `apply()`.